handbook/engineering/cloud/index.md

```diff
@@ -2,32 +2,53 @@
 
 The cloud team owns all work that is necessary to build, secure, scale, and operate our multi-tenant hosted version of Sourcegraph for customers that do not want to deploy Sourcegraph on-premise.
 
+The cloud team is also responsible for all [backend-infrastructure areas of ownership](../backend-infrastructure/index.md). A lot of the work for Sourcegraph Cloud is in direct support of scaling and performance of backend-infrastructure. Work will be balanced as needed to support both efforts.
+
+## Areas of ownership
+
+- Authorization and authentication
+- Repository management (gitserver, repo-updater, src-expose)
+- Data storage and access libraries
+- GraphQL API
+- src-cli
+- License generation and enforcement
+- Admin and user settings
+- Analytics
+
 ## Goals
 
+### Sourcegraph Cloud
+
 1. Any user or organization can use Sourcegraph Cloud with both the private and public code that they care about in a secure way.
 
-## Milestones
+#### Milestones
 
-1. Any user can add public code from github.com, gitlab.com and bitbucket.org
+1. The Sourcegraph organization and team members can add public code from GitHub.com, GitLab.com, and Bitbucket Cloud.
     - Code is indexed.
     - It's easy to search over only their code (vs all public code).
```

Comment:
@christinaforney: We added this previously because we think it's going to be important. There are multiple solutions here, but this feels like the domain of the search team. We can also punt this to the "make it usable" phase. Is there a good way for you to capture this work for that phase in the search team's roadmap?

Reply:
I agree this is is going to be important - and I think earlier in the process than later. As for punting it, perhaps having two versions of this: 1. it is possible to search over only your code (even if it's a janky implementation that our team can use), and then make it usable is more about having a clear way of knowing what you're searching over.

It's a good question, and one I would like to loop @lguychard into. I am wondering if it makes sense to put this on the search team, or if it makes sense for the Cloud team to own because it's a requirement to achieve the team's goals. I was previously assuming that the Cloud team would own some cross functional efforts, but if not, we should also talk about anything else that is required from other teams. Does this also mean that the web team should be planning to implement the UX for these efforts? Where do we draw the line and how do we make it clear to everyone?

Reply:
> As for punting it, perhaps having two versions of this (..) (even if it's a janky implementation that our team can use)

Yeah, I think that's possible. One idea: we tie in to repo groups, and create a "special" repogroup called `default` or `mine`, that refers to repos the user has added. Thoughts on this approach @sourcegraph/search?

>  Where do we draw the line and how do we make it clear to everyone

In my mind, if there's another team that owns the end product user experience of a particular area, then they need to be 100% aligned with or driving an effort in that area.

Examples: 
- Search team owns the search experience, so we need them to be 100% aligned with or driving the effort of making it easy for users to search over their public code.
- Cloud team owns admin and user settings, so we can drive changes to the user experience independently of other teams.

I'm happy to discuss this and align on whatever collaboration model we see fit, but this is how I see it.

Reply:
Based on priorities/resources, it is likely the Cloud team will need to do the implementation, and I agree there needs to be close alignment/collaboration on planning/architecture with the search team.

Reply:
Having two versions sounds like a great idea:
- The Cloud team could execute on the "MVP" to unblock this goal.
- The Search team would be better equipped to tackle the end target ("users have a clear way of knowing / filtering what they're searching over, including filtering to search over their own code").

On the more general point:

> if there's another team that owns the end product user experience of a particular area, then they need to be 100% aligned with or driving an effort in that area

The distinction between "100% aligned" and "driving an effort" depends on the size of the effort needed. For small, self-contained efforts like the "special repogroup", aligning with the search team on the solution but having the Cloud team execute is probably the quickest way of moving forward.

It gets trickier when the effort is large enough that the other team would have to "drive the effort in that area" -- dependencies between teams are fragile, and I find it better for teams to be autonomously capable of making progress on their goals.

If a goal is truly large enough in scope that it would require effort from multiple teams, then it's a company goal -- we should then align _all teams_ around it, and structure/break up the execution strategy so that teams can make progress towards that goal without being mutually dependent on each other.

If a team finds itself blocked on making progress towards their goal for lack of a given skillset (eg. frontend skills), it would be more efficient to fill that gap through hiring or internal transfer than through delegation to another team.

Decoupling team priorities will keep teams focused on and engaged with business problems rather than organizational problems: "we're prioritizing this because it's the right thing to do to solve problem X" rather than "we're prioritizing this to unblock team Y".